src/tools/server.ml

```diff
@@ -209,6 +209,7 @@ let make_server_thread
         Lwt.catch
           (fun () ->
            Lwt_unix.accept listening_socket >>= fun (plain_fd, cl_socket_address) ->
+           Lwt_unix.setsockopt listening_socket Unix.SO_KEEPALIVE true;
```

Comment:
yes, something like that.Actually, we don't even need to modify the code to make this available: you can add it to your startup wrapper via library preloading (warning: I never actually tried this)